### PR TITLE
show: Throw clear error message on unsupported kernels

### PR DESCRIPTION
**Pull request by velurimithun**
'schemes_tried_regions' is only available from v6.2

Fixes issue #101

*Description of changes:*

Env:
```
[root@ip-172-31-7-19 damo]# uname -r
6.1.82-99.168.amzn2023.x86_64
[root@ip-172-31-7-19 damo]# damo start 
[root@ip-172-31-7-19 damo]# damo features | grep schemes_tried_regions
schemes_tried_regions: Unsupported
schemes_tried_regions_sz: Unsupported
[root@ip-172-31-7-19 damo]# 
```

Before:
```
[root@ip-172-31-7-19 damo]# damo show
updating schemes tried regions fail: writing update_schemes_tried_regions to /sys/kernel/mm/damon/admin/kdamonds/0/state failed ([Errno 22] Invalid argument) (maybe schemes_tried_regions not supported?)
[root@ip-172-31-7-19 damo]# 
```

After:

```
[root@ip-172-31-7-19 damo]# damo show
updating schemes tried regions fail: writing update_schemes_tried_regions to /sys/kernel/mm/damon/admin/kdamonds/0/state failed ([Errno 22] Invalid argument) (DAMON feature 'schemes_tried_regions' is supported only from kernel version 6.2)
[root@ip-172-31-7-19 damo]# 
```

*Testing:*
```
[root@ip-172-31-7-19 damo]# ./tests/run.sh 
PASS unit test_damo_records.py
PASS unit test_damo_scheme_dbgfs_conversion.py
PASS unit test_damo_schemes_input.py
[...]
[...]
PASS start_stop sysfs stop
PASS start_stop
PASS ALL
[root@ip-172-31-7-19 damo]#
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/_damon_sysfs.py [hunk 1]
```diff
@@ -74,7 +74,8 @@ def update_schemes_tried_bytes(kdamond_idxs):
 def update_schemes_tried_regions(kdamond_idxs):
     err = __write_state_file(kdamond_idxs, 'update_schemes_tried_regions')
     if err != None:
-        err = '%s (maybe schemes_tried_regions not supported?)' % err
+        # schemes_tried_regions feature available from mainline kernel v6.2 only
+        err = '%s (DAMON feature \'schemes_tried_regions\' is supported only from kernel version 6.2)' % err
```
Comment (sjp38) [inline src/_damon_sysfs.py:78]: `__wite_state_file()` writes the keyword to the `state` file.  Writing to `state` file can fail if the there is a concurrent thread that doing another write already.  Hence, the reason of the failure could also be the concurrent writes.  What about keeping the original message but appending this hint? ("maybe schemes_tried_regions not supported?  AMON feature \'schemes_tried_regions\' is supported only from kernel version 6.2")

**Comment by sjp38** [comment]
Thank you for this nice PR!  I added a question to the diff.

**Comment by sjp38** [comment]
Also, let's drop "Fix/" prefix from the commit's subject.  I think it could make the commit looks not consistent with other commits.

File: src/_damon_sysfs.py (hunk 1 above)
Comment (velurimithun) [reply src/_damon_sysfs.py:78]: ok got it, in that case, I was thinking of checking the current running kernel version and throwing this error (`DAMON feature \'schemes_tried_regions\' is supported only from kernel version 6.2`) only in cases when kernel version if < 6.2. In all other cases, we will just throw error (`updating schemes tried regions fail: writing update_schemes_tried_regions to /sys/kernel/mm/damon/admin/kdamonds/0/state failed ([Errno 22] Invalid argument)`) we receive from `__write_state_file `.

My idea is to not leave a question (`maybe schemes_tried_regions not supported?`) for customer, what do you say about this approach?

Comment (sjp38) [reply src/_damon_sysfs.py:78]: I think removing the question is a nice approach.  Nonetheless, some distros provide kernels that having DAMON future features backported.  For example, AL2 provides v5.10 based kernel that having v6.7's DAMON code.  So, the error code may better to be adjusted for such cases.

I think we can use `feature_supported()` function to check if the running kernel is supporting the `update_schemes_tried_regions` and provide an error message.

Comment (velurimithun) [reply src/_damon_sysfs.py:78]: Sure SJ, I'll use `feature_supported()` to check if the running kernel supports `update_schemes_tried_regions ` and provide error message accordingly.

Comment (sjp38) [reply src/_damon_sysfs.py:78]: Thanks, looking forward to the next revision!

Comment (velurimithun) [reply src/_damon_sysfs.py]: Sent a revision.
Used `feature_supported ()` to check if the feature is supported and accordingly throwing error message.

**Pull request merged**